src/webots/nodes/utils/WbNodeUtilities.cpp

```diff
@@ -758,17 +762,43 @@ bool WbNodeUtilities::isDescendantOfBillboard(const WbNode *node) {
     return true;
 
   WbNode *n = node->parentNode();
-  WbField *field = node->parentField(true);
-  while (n && !n->isWorldRoot() && field) {
-    WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(field->parentNode());
+  while (n && !n->isWorldRoot()) {
+    WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(n);
 
     if (!baseNode)
       return false;
 
     if (baseNode->nodeType() == WB_NODE_BILLBOARD)
       return true;
 
-    field = n->parentField(true);
+    n = n->parentNode();
+  }
+
+  return false;
+}
+
+bool WbNodeUtilities::isDescendantOfPropeller(const WbNode *node) {
+  if (node == NULL)
+    return false;
+
+  const WbBaseNode *initialNode = dynamic_cast<const WbBaseNode *>(node);
+
+  if (!initialNode)
+    return false;
+
+  if (initialNode->nodeType() == WB_NODE_PROPELLER)
+    return true;
+
+  WbNode *n = node->parentNode();
+  while (n && !n->isWorldRoot()) {
+    WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(n);
+
+    if (!baseNode)
+      return false;
+
+    if (baseNode->nodeType() == WB_NODE_PROPELLER)
+      return true;
+
     n = n->parentNode();
   }
```

Comment:
You could also further simplify this:
```suggestion
  WbNode *n = node;
  while (n && !n->isWorldRoot()) {
    const WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(n);
    if (!baseNode)
      return false;
    if (baseNode->nodeType() == WB_NODE_PROPELLER)
      return true;
    
    n = n->parent();
  }
```